network/omaha/server.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package omaha
+
+import (
+	"encoding/xml"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/coreos/mantle/Godeps/_workspace/src/gopkg.in/tylerb/graceful.v1"
+)
+
+func NewServer(addr string) (*Server, error) {
+	l, err := net.Listen("tcp", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	srv := &graceful.Server{
```

Comment:
I'm guessing we want to set `NoSignalHandling` to true, seems bad form for libraries to go clobering signal handlers as this one appears to do.